### ISSUE TITLE
Refresh meter design

### ISSUE THREAD

**Issue by dominicduffin1** [prog-o-meter/prog-o-meter#95]
I'd like to suggest some changes to the design of the meter:

1. Make canvas background white
2. Change the initial fill colour of the rectangles to a light grey
3. Change the outline of the rectangles to white to match the new canvas background colour
4. Change the shade of the blue and green to a fresher custom shade

If these ideas sound good, I'd like to work on it. If not, I'm happy to change the ideas.

**Comment by lineaba** [comment]
It all sounds good. I have one "worry", one question and one comment:
What do you mean with number 3? If we make the outlines white on white background, would that mean that each individual rectangle would not be visible, or would they still be visible, because their basic color is light grey?
My "worry" is whether there will still be coherence between the welcome and log-in window which have grey background, and then the progress window, which will be white? 
The comment: I just wanted to mention that we are working towards personalization where each user can choose what color they want for progressed days (to replace the green/blue), but we can definitely still change the default colors to other shades. We might wanna consider using hex, and testing it on some different platforms, because I know that currently, the shades of blue/green actually differs based on the user's operating system.

**Comment by dominicduffin1** [comment]
With number 3, yes, the grey background means they will still be visible, but the borders will blend with the canvas.
Coherence is a potential issue. One idea I had that might help is to put the buttons in an area to the side of the canvas, which would have a grey background the same as the other windows. This would also have plenty of room for some more buttons for user customisation options.
Using hex would be a good idea.
Working towards personalisation is good - we could also work towards allowing the user to customise the canvas background and rectangle borders - I'm all for allowing the user make things the way they prefer.

**Comment by lineaba** [comment]
A button panel is a nice idea, I could imagine that working out. The only question is whether it should be to a side, or maybe rather in the bottom of the window. Either way, we wanna make sure that we keep room for more buttons, but I imagine that the personalisation will only have one button, which will take the user into a customization setting window. (However, I am open to other solutions here). 
Regarding grid over pack, I don't have an opinion, I have never used grid. Would you mind sharing some thoughts about why you think it would be a better idea?
Regarding personalisation, I think it is a good idea to give the users some options, but we also don't wanna overwhelm them with too many options I think. It is of course a balance to find out how many are too many.

**Comment by lineaba** [comment]
Just an update, @etiontdn have already been working on some personalisation letting the user choose colors etc. The code is not merged yet, just wanted to let you know that it is on its way.

**Comment by dominicduffin1** [comment]
I think it would probably be best to put the button panel at the bottom of the window. I personally prefer the aesthetics of the side option, but on reflection I think it will work better with the constraint of the shape of the meter (which we definitely don't want to change).
My thinking about using grid is that when you've got quite a few elements to position it's easier to create a nice layout by working with rows and columns.

**Comment by lineaba** [comment]
Yes, the shape of the meter gives some challenges, and it will be interesting how to handle that when developing a webapp in the future, given that webpages are often used on phones (but that is a worry for later, and not relevant to your question. I agree that button panel at the bottom is the way to go.
Regarding grid, I don't see any issues with using it, giving that we are already using pack and place pretty interchangeably (using pack or place really just depends on who wrote the given code). Of course it might be confusing for new contributors (especially if they are new to tkinter that we aren't using any method consistently, and this is something we could potentially adress at a later stage. 
Long answer short, go ahead with grid, if you believe that is the best solution.

**Comment by dominicduffin1** [comment]
Do we have any idea on the timescale for the personalisation changes? Just wondering whether it would be best to wait for them to be implemented before starting any redesign.

**Comment by etiontdn** [comment]
@dominicduffin1 The GUI for personalization that i was working on didn't get a code review yet, but it is getting out of date from the recent commits, i don't have a lot of time for working on it either, you can take a look at it on my fork if you want.
I can update it later if you want to work on design it is outdated already, updating a little more won't change the things a lot. 😃
